flink-clients/src/main/java/org/apache/flink/client/program/ContextEnvironment.java

```diff
@@ -91,25 +84,35 @@ public JobExecutionResult execute(String jobName) throws Exception {
 			}
 
 			jobExecutionResult = jobExecutionResultFuture.get();
```

Comment:
@aljoscha I notice that we don't callback job listener if context environment calls...we don't call `super.execute` here.

Reply:
Good catch! Yes, we also need to fix that.